core/corehttp/gateway_handler.go

```diff
@@ -854,7 +854,7 @@ func webError(w http.ResponseWriter, message string, err error, defaultCode int)
 func webErrorWithCode(w http.ResponseWriter, message string, err error, code int) {
 	http.Error(w, fmt.Sprintf("%s: %s", message, err), code)
 	if code >= 500 {
-		log.Warnf("server error: %s: %s", err)
+		log.Warnf("server error: %s: %s", message, err)
```

Comment:
The linter is complaining about this being unsanitized. IIUC these are all escaped URLs, right @lidel?

Reply:
@lidel, @guseggert, @MichaelMure if any of you get a chance to confirm the linter is off here please comment and anyone with merge access can come by and stamp it.

Thanks @MichaelMure for catching this 🙏.

Reply:
The linter is right:
- https://github.com/ipfs/go-ipfs/blob/cbb962852f914dc682887ea86c227e8f546c880c/core/corehttp/gateway_handler.go#L301
- https://github.com/ipfs/go-ipfs/blob/cbb962852f914dc682887ea86c227e8f546c880c/core/corehttp/gateway_handler.go#L200

Reply:
Does the path escaping from `EscapedPath` not help us out here, and we should be doing our own escaping?

Reply:
I can't say for sure, but this escaping looks more about encoding for a URL path rather than security.

Reply:
- klinda false-positive: `webError` is returned as `text/plain` so it does not need to be protected against HTML injection (we don't have to wrap it in HTMLEscapeString)